File: 03.programmers/lv02/02.Python/07_develop_function.py
def solution(progresses, speeds):
    answer = []
    max_day = 0
    for p, s  in zip(progresses, speeds):
        day = (99 - p)//s +1
        if len(answer) == 0 or max_day< day:
            max_day = day
            answer.append(1)
        else: answer[-1]+=1

    return answer

input_progresses = [93, 30, 55]
input_speeds = [1, 30, 5]
print(solution(input_progresses, input_speeds))

File: ~2023/03.programmers/lv02/01.C++/07_develop_function.cpp
#include <iostream>
#include <string>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int max_day = 0;
    for (int i = 0; i < progresses.size(); ++i){
        int day = (99 - progresses[i]) / speeds[i] + 1;
        if (answer.empty() || max_day < day) {
            answer.push_back(1);
            max_day = day;
        }
        else answer.back()++;
    }
    return answer;
}
int main(){
    // vector<int> p = {40, 93, 30, 55, 60, 65};
    vector<int> p = {93,30,55};
    // vector<int> s = {60, 1, 30, 5 , 10, 7};
    vector<int> s = {1,30,5};
    for(auto ans : solution(p,s))
        cout<<ans<<" ";
    cout<<endl;
    return 0;
}